### PR TITLE
Warn if user not in multisig

### LINKED ISSUE #10: Raise error if user is not part of the multisig
Right now I can sign a transaction for a multisig that I am not part of. It even shows "1 of 2 signatures complete" in that case. This is not dangerous because my signature is just worthless. However, it would be nice to tell the user earlier to avoid a hard to debug signature verification failure later on.

This case can happen easily if a user has multiple accounts.

Bildschirmfoto 2022-01-10 um 16 34 50

### PR DESCRIPTION
**Pull request by abefernan**
Closes #10

### CODE REVIEW

File: components/forms/TransactionSigning.tsx [hunk 1]
```diff
@@ -21,15 +23,18 @@ interface LoadingStates {
 interface Props {
   signatures: DbSignature[];
   tx: DbTransaction;
+  pubkey: MultisigThresholdPubkey;
   transactionID: string;
   addSignature: (signature: DbSignature) => void;
 }
 
 const TransactionSigning = (props: Props) => {
+  const memberPubkeys = props.pubkey?.value.pubkeys.map(({ value }) => value);
```
Comment (webmaster128) [inline components/forms/TransactionSigning.tsx:32]:
```suggestion
  const memberPubkeys = props.pubkey.value.pubkeys.map(({ value }) => value);
```

this is not optional, right?

File: types/index.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export interface DbAccount {
 
 export interface WalletAccount {
   address?: Uint8Array;
-  pubkey: Uint8Array;
+  pubKey: Uint8Array;
```
Comment (webmaster128) [inline types/index.ts:42]: Nice catch

File: components/forms/TransactionSigning.tsx [hunk 3]
```diff
@@ -44,12 +49,23 @@ const TransactionSigning = (props: Props) => {
...
+        : false;
+      if (hasSigned) {
+        setSigning("signed");
+      }
+      if (isMember && !hasSigned) {
+        setSigning("not_signed");
+      }
+      if (!isMember) {
+        setSigning("not_a_member");
+      }
```
Comment (webmaster128) [inline components/forms/TransactionSigning.tsx:67]: You could write those something like this to ensure `setSigning` is only called one time. But I don't mind much as the code seems to be correct.

```suggestion     
      if (!isMember) {
        setSigning("not_a_member");
      } else if {
           if (hasSigned) {
                setSigning("signed");
          } else {
                setSigning("not_signed");
          }
      }
```

**Review by webmaster128 — COMMENTED** [review]
👏

File: components/forms/TransactionSigning.tsx (hunk 1 above)
Comment (abefernan) [reply components/forms/TransactionSigning.tsx:32]: It is undefined when the component loads and then gets a value. Maybe the type is wrong

Comment (webmaster128) [reply components/forms/TransactionSigning.tsx:32]: okay, but it seems like you already cover that case in the caller code. See diff in `pages/multi/[address]/transaction/[transactionID].tsx` below

**Pull request merged**